### DISCUSSION TITLE
Automatic removal of dependencies that are not longer needed

### DISCUSSION THREAD

**Discussion by jborbik** [Homebrew/discussions#2090] [Everyday usage] [answered]
Hello,

Is there any way to enforce automatic cleanup of the dependencies which are not longer needed by any of the formula? I have an impression that many of the installs became leaves in my case after they were removed from dependencies from my requested installs.

**Comment by carlocab** [comment, marked as answer]
If you use `brew bundle` you can do `brew bundle cleanup --force` to remove installed formulae you no longer need. You'll need to have a `Brewfile` that specifies the formula you want to be installed, though. See https://github.com/Homebrew/homebrew-bundle.

If you don't want to create/maintain a `Brewfile`, you can use `brew autoremove`.  You may want to do `brew autoremove --dry-run` first to make sure it's not removing anything you still want to keep. See `brew help autoremove` for details.

Finally, you can also use `brew leaves` to see your installed formulae with no dependencies, as these can be safely removed if they're no longer being used.

**Reply by jborbik** [reply]
Thank you so much for the complete answer!
